### DISCUSSION TITLE
Formatted docstring (leading f) does not show up in API documentation

### DISCUSSION THREAD

**Discussion by JRGit4UE** [fastapi/fastapi#11193] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

detected_support = "capabilities requested from other service"

app = FastAPI()


@app.get("/")
async def read_root():
    f"""This is the glorious documentation of read_root. It supports {detected_support}."""
    return {"Hello": "World"}


if __name__ == "__main__":
    import uvicorn
    uvicorn.run(app)
```

### Description

1) Open the browser and browse to endpoint /docs.  I can`t see any API documentation from the block comment for /.  
2) Remove the leading  `f` from the block comment and restart the server.
3) Now, you see the documentation with `{detected_support}`.

### Operating System

Linux

### Operating System Details

* Ubuntu 22 or Windows 10

### FastAPI Version

fastapi==0.110.0

### Pydantic Version

2.6.2

### Python Version

3.10

**Comment by jgould22** [comment, marked as answer]
f Strings are not doc strings.

Python will not place an f string in the `__doc__` attribute which is what is used for the docs.

You can work around this by setting the `__doc__` attribute yourself but it isnt super elegant.

**Comment by JRGit4UE** [comment]
@jgould22 
Thank you for your solution, I was not aware of docstring handling so far. 
> ... it isn't super elegant

**d'accord**  but in the case that a service depends on other services, I would find it incredibly convenient if my API documentation would reflect potential changes, so that people save a lot of time in the try-out phase. 

I have tried to set the `__doc__` property of the `read_root()` function, via
``` python
@app.get("/")
async def read_root():
    """This is the default documentation."""
    return {"Hello": "World"}

if necessary:
    read_root.__doc__= f"This is the {updated_documentation}"
```
.. but it does not work that way.  
Do you by chance know how to set the docstring of a decorated API function?
Thxalot

**Reply by Kludex** [reply]
You need to register the `read_root` to the app after the changes of the `__doc__`.

```py
from fastapi import FastAPI

app = FastAPI()


async def read_root():
    """This is the default documentation."""
    return {'Hello': 'World'}


if True:
    updated_documentation = 'updated documentation'
    read_root.__doc__ = f'This is the {updated_documentation}'

app.get('/')(read_root)
```

> [!important]
> I'm not recommending this.

**Reply by JRGit4UE** [reply]
Thank you for your help!
... and I am aware about the discussions on this topic (e.g. [here](https://softwareengineering.stackexchange.com/questions/327578/is-it-poor-practice-to-have-dynamically-updated-docstrings)).
